extensions-core/druid-catalog/src/main/java/org/apache/druid/catalog/http/CatalogResource.java

```diff
@@ -0,0 +1,747 @@
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response moveColumn(
+      @PathParam("dbSchema") final String dbSchema,
+      @PathParam("name") final String name,
+      final MoveColumn command,
+      @Context final HttpServletRequest req
+  )
+  {
+    if (command == null) {
+      return Actions.badRequest(Actions.INVALID, "A MoveColumn object is required");
+    }
+    if (Strings.isNullOrEmpty(command.column)) {
+      return Actions.badRequest(Actions.INVALID, "A column name is required");
+    }
+    if (command.where == null) {
+      return Actions.badRequest(Actions.INVALID, "A target location is required");
+    }
+    if ((command.where == Position.BEFORE || command.where == Position.AFTER) && Strings.isNullOrEmpty(command.anchor)) {
+      return Actions.badRequest(Actions.INVALID, "A anchor column is required for BEFORE or AFTER");
+    }
+    return incrementalUpdate(
+        TableId.of(dbSchema, name),
+        null,
+        req,
+        (spec) -> spec.withColumns(command.perform(spec.columns()))
+    );
+  }
+
+  /**
+   * Hide or unhide columns. If both appear, hide takes precedence. Returns the
+   * new table version.
+   */
+  @POST
+  @Path("/tables/{dbSchema}/{name}/hideColumns")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response hideColumns(
+      @PathParam("dbSchema") final String dbSchema,
+      @PathParam("name") final String name,
+      final HideColumns command,
+      @Context final HttpServletRequest req
+  )
+  {
+    return incrementalUpdate(
+        TableId.of(dbSchema, name),
+        null,
+        req,
+        (spec) -> {
+          if (!AbstractDatasourceDefn.isDatasource(spec.type())) {
+            throw new ISE("hideColumns is supported only for data source specs");
+          }
+          @SuppressWarnings("unchecked")
+          List<String> hiddenProps = (List<String>) spec.properties().get(AbstractDatasourceDefn.HIDDEN_COLUMNS_PROPERTY);
+          return spec.withProperty(
+              AbstractDatasourceDefn.HIDDEN_COLUMNS_PROPERTY,
+              command.perform(hiddenProps)
+          );
+        }
+    );
+  }
+
+  /**
+   * Drop column metadata. Only removes metadata entries, has no effect on the
+   * physical segments. Returns the new table version.
+   */
+  @POST
+  @Path("/tables/{dbSchema}/{name}/dropColumns")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response dropColumns(
+      @PathParam("dbSchema") final String dbSchema,
+      @PathParam("name") final String name,
+      final List<String> columns,
+      @Context final HttpServletRequest req
+  )
+  {
+    return incrementalUpdate(
+        TableId.of(dbSchema, name),
+        null,
+        req,
+        (spec) -> spec.withColumns(dropColumns(spec.columns(), columns))
+    );
+  }
+
+  private static <T extends ColumnSpec> List<T> dropColumns(
```

Comment:
Should move this into a class as is done with HideColumns and MoveColumns to be consistent.

Reply:
ah gotcha ok, makes sense.